src/VisualStudio/Core/Def/Implementation/ProjectSystem/FileChangeWatcher.cs

```diff
@@ -189,14 +189,7 @@ public Context(FileChangeWatcher fileChangeWatcher, ImmutableArray<WatchedDirect
 
                             if (watchedDirectory.ExtensionFilter != null)
                             {
-                                // TODO: switch to proper reference assemblies
-                                var filterDirectoryChangesAsyncMethod = service.GetType().GetMethod("FilterDirectoryChangesAsync");
-
-                                if (filterDirectoryChangesAsyncMethod != null)
-                                {
-                                    var arguments = new object[] { cookie, new string[] { watchedDirectory.ExtensionFilter }, CancellationToken.None };
-                                    await ((Task)filterDirectoryChangesAsyncMethod.Invoke(service, arguments)).ConfigureAwait(false);
-                                }
+                                await service.FilterDirectoryChangesAsync(cookie, new string[] { watchedDirectory.ExtensionFilter }, CancellationToken.None).ConfigureAwait(false);
```

Comment:
Oh nice, I just got bitten by this reflection today morning: https://github.com/dotnet/roslyn/pull/45989/commits/5a7af674839395840fc064bd8ce8290dc65662ff

Reply:
Hah! Note this is going into -vs-deps; whenever that change makes it into master let me know so I can make sure it's fixed up right in -vs-deps.